src/daily-dropoffs-report.tsx

```diff
@@ -0,0 +1,392 @@
+      bin.forEach((row) => {
+        STEPS.forEach(({ key }) => {
+          const oldCount = steps.get(key) || 0;
+          steps.set(key, (row[key] || 0) + oldCount);
+        });
+      });
+
+      const { issuer, friendly_name: friendlyName, iaa, agency, start, finish } = bin[0];
+
+      return {
+        issuer,
+        friendly_name: friendlyName,
+        iaa,
+        agency,
+        start,
+        finish,
+        ...Object.fromEntries(steps),
+      } as DailyDropoffsRow;
+    });
+}
+
+function tabulate({
+  rows: results,
+  filterAgency,
+  issuerColor,
+}: {
+  rows?: DailyDropoffsRow[];
+  filterAgency?: string;
+  issuerColor: (issuer: string) => string;
+}): TableData {
+  const filteredRows = (results || []).filter((d) => !filterAgency || d.agency === filterAgency);
+
+  const header = [
+    "Agency",
+    "App",
+    ...STEPS.map(({ title }, idx) => <th colSpan={idx === 0 ? 1 : 2}>{title}</th>),
+  ];
+
+  const color = scaleLinear()
+    .domain([1, 0])
+    .range([
+      // scaleLinear can interpolate string colors, but the 3rd party type annotations don't know that yet
+      "steelblue" as unknown as number,
+      "white" as unknown as number,
+    ]);
+
+  const body = filteredRows.map((row) => {
+    const { agency, issuer, friendly_name: friendlyName } = row;
+
+    return [
+      agency,
+      <span title={issuer}>
+        <span style={`color: ${issuerColor(issuer)}`}>⬤ </span>
+        {friendlyName}
+      </span>,
+      ...toStepCounts(row).flatMap(({ count, percentOfFirst }, idx) => {
+        const backgroundColor = `background-color: ${color(percentOfFirst)};`;
+        const cells = [
+          <td className="table-number text-tabular text-right" style={backgroundColor}>
+            {formatWithCommas(count)}
+          </td>,
+        ];
+
+        if (idx > 0) {
+          cells.push(
+            <td className="table-number text-tabular text-right" style={backgroundColor}>
+              {formatAsPercent(percentOfFirst)}
+            </td>
+          );
+        }
+
+        return cells;
+      }),
+    ];
+  });
+
+  return {
+    header,
+    body,
+  };
+}
+
+function loadData(
+  start: Date,
+  finish: Date,
+  env: string,
+  fetch = window.fetch
+): Promise<DailyDropoffsRow[]> {
+  return Promise.all(
+    utcDays(start, finish, 1).map((date) => {
+      const path = reportPath({ reportName: "daily-dropoffs-report", date, env, extension: "csv" });
+      return fetch(path).then((response) => response.text());
+    })
+  ).then((reports) => aggregate(reports.flatMap((r) => process(r))));
+}
+
+function Axis({
+  axis,
+  transform,
+  rotateLabels,
+  className,
+}: {
+  axis: D3Axis<NumberValue> | D3Axis<string>;
+  transform: string;
+  rotateLabels?: boolean;
+  className?: string;
+}): VNode {
+  const ref = useRef(null as SVGGElement | null);
+
+  useEffect(() => {
+    if (ref.current) {
+      select(ref.current).call(axis).classed("rotate-labels", !!rotateLabels);
+    }
+  });
```

Comment:
We should include dependencies here, even if it's just an empty array. Otherwise, `useEffect` is called on every render.

```suggestion
  }, [axis, rotateLabels]);
```